conan/tools/gnu/pkgconfigdeps.py

```diff
@@ -209,8 +234,9 @@ def shortened_content(self, info):
         assert isinstance(info, _PCInfo)
         custom_content = info.cpp_info.get_property("pkg_config_custom_content") if info.cpp_info \
             else None
+        prefix_path = "" if self._is_var_match("prefix", custom_content) else self._get_prefix_path()
```

Comment:
IMHO we should leave the prefix path out of this logic as it contains the package folder? If we want users to be able to define prefix in the custom content, another option would be to not use the prefix at all and add the absolute paths directly in our pc files?

Reply:
I don't think so. `prefix` is part of the freeform variables added to all the *.pc files so if a user decides to overwrite it, let's allow it. I think it won't ever happen, but it's so, it would be a similar case if they decide to override the `package_folder` or any other variable defined by Conan, wouldn't it?